cocos/base/ObjectFactory.h

```diff
@@ -41,14 +41,11 @@ class CC_DLL ObjectFactory
     typedef std::function<cocos2d::Ref* ()> InstanceFunc;
     struct CC_DLL TInfo
     {
-        TInfo();
+        TInfo() = default;
         TInfo(const std::string& type, Instance ins = nullptr);
```

Comment:
I think that default =nullptr should be removed for both constructors.

Reply:
Yeah. It causes ambiguous overload resolution.
```cpp
TInfo(const std::string& type, Instance ins = nullptr);
TInfo(const std::string& type, const InstanceFunc& ins = nullptr);
```

The following code wouldn't compile because of it.

```cpp
int main(int argc, char **argv)
{
    ObjectFactory::TInfo t("classname");
}

Need to remove default parameters from them.
```